src/ReferenceList.php

```diff
@@ -126,7 +126,7 @@ public function hasReference( Reference $reference ) {
 	 */
 	public function indexOf( Reference $reference ) {
 		foreach ( $this->references as $index => $ref ) {
-			if ( $ref->equals( $reference ) ) {
+			if ( $ref === $reference ) {
```

Comment:
This is now the same behaviour as before. I can create a test for the old behaviour though.